### PR TITLE
Update Javadoc

### PR DESCRIPTION
**Pull request by brfrn169**
Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/api/DistributedStorage.java [hunk 1]
```diff
@@ -31,31 +31,45 @@
  * readability.)
  *
  * <pre>{@code
- * StorageFactory factory = new StorageFactory(databaseConfig);
+ * StorageFactory factory = StorageFactory.create(configFilePath);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedStorage.java:34]: Fixed this to use the new API.

File: core/src/main/java/com/scalar/db/api/DistributedStorageAdmin.java [hunk 2]
```diff
@@ -10,13 +10,15 @@
  * readability.)
  *
  * <pre>{@code
- * StorageFactory factory = new StorageFactory(databaseConfig);
+ * StorageFactory factory = StorageFactory.create(configFilePath);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedStorageAdmin.java:13]: Ditto.

File: core/src/main/java/com/scalar/db/api/TransactionCrudOperable.java [hunk 3]
```diff
@@ -5,7 +5,11 @@
 import java.util.List;
 import java.util.Optional;
 
-/** An interface for transaction CRUD operations. */
+/**
+ * An interface for transaction CRUD operations. Note that LINEARIZABLE consistency is always used
+ * in transactional CRUD operations, so {@link Consistency} specified for CRUD operations is
+ * ignored.
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/TransactionCrudOperable.java:11]: Add a note that LINEARIZABLE consistency is always used in transactional CRUD operations.

File: core/src/main/java/com/scalar/db/api/DistributedStorage.java [hunk 4]
```diff
@@ -31,31 +31,45 @@
...
  * Optional<Result> result = storage.get(get);
  *
  * // Deletes an entry which has the primary key value 1.
- * Delete delete = new Delete(new Key(COL_NAME, 1);
+ * Delete delete =
+ *     Delete.newBuilder()
+ *         .namespace(NAMESPACE)
+ *         .table(TABLE)
+ *         .partitionKey(Key.ofInt(COL_NAME, 0))
+ *         .build();
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedStorage.java:64]: Fixed this to use the new API.

File: core/src/main/java/com/scalar/db/api/DistributedStorageAdmin.java (hunk 2 above)
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedStorageAdmin.java:13]: Ditto.

File: core/src/main/java/com/scalar/db/api/TransactionCrudOperable.java [hunk 5]
```diff
@@ -77,7 +81,8 @@ public interface TransactionCrudOperable {
 
   /**
    * Mutates entries of the storage through a transaction with the specified list of {@link
-   * Mutation} commands.
+   * Mutation} commands. Note that the conditions set in Mutation will be ignored. Please program
+   * such conditions in a transaction if you want to implement conditional mutation.
```
Comment (Torch3333) [inline core/src/main/java/com/scalar/db/api/TransactionCrudOperable.java:85]: Sorry, I don't understand how to set mutation in a transaction? You are not referring to `Put.withCondition` method right ?

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!
I left a single question, please have a look.

File: core/src/main/java/com/scalar/db/api/TransactionCrudOperable.java (hunk 5 above)
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/api/TransactionCrudOperable.java:85]: I meant even if you set conditions to the specified mutations (with Put.withCondition() and Delete.withCondition() method), they will be ignored in transactions.

Actually, this comment is already in the Javadocs for `put()` and `delete()`, but we missed it for `mutate()`, so I added it.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**